### DISCUSSION TITLE
Pushover and Temperature Notification

### DISCUSSION THREAD

**Discussion by McBlade** [nebhead/PiFire#65] [Ideas]
Also, here, thanks for this project. It is great and I like to playing with.

One idea for the pushover and maybe other notification. Maybe it make sense to implement a test button? I am doing the construction of the electronic on my desk and test the different things. So it will be easier to test the connection of the notification methods, if there is a test button, that shows that the connection is ok.

Is it correct, that the notification of the temperature, is only working if the system is in working mode? Maybe could be interesting to have it in Monitor Mode, too.

Do you think it is possible to implement a second temperature notification for the grill probe? Could be usefull, if due to some reason the temperature going down, under a certain level.

This are just ideas. I am very happy, that I have found your project and can work with. I am looking forward to build my pellet hooper (I will do it on my own completely) and cook with it in summer.

Thank you

Stefan

**Comment by nebhead** [comment]
> Also, here, thanks for this project. It is great and I like to playing with.
> 
> One idea for the pushover and maybe other notification. Maybe it make sense to implement a test button? I am doing the construction of the electronic on my desk and test the different things. So it will be easier to test the connection of the notification methods, if there is a test button, that shows that the connection is ok.

Oh that's an interesting idea, yeah, could be something we look into for future testing.  If you just want to test it out on your bench, I'd recommend running PiFire with the prototype grill & adc modules installed.  This will allow you to simulate the functionality, so that you can test all the features.  If you're running Linux on your desktop/development machine, you can even run PiFire, using the prototype modules, on your desktop.  You can run `sudo bash modules.sh` from the command line to configure the prototype modules.  You'll have to restart the control.py script after making these changes either through supervisor or by rebooting.  

> Is it correct, that the notification of the temperature, is only working if the system is in working mode? Maybe could be interesting to have it in Monitor Mode, too.

Notifications should work in monitor mode as well.

> Do you think it is possible to implement a second temperature notification for the grill probe? Could be usefull, if due to some reason the temperature going down, under a certain level.

Notifications are working for grill probes as well.  Just click the bell icons in the WebUI on each of the probes.  This will give you a notification once the set temperature is achieved.  

I made a video that covers most of the features [here](https://youtu.be/_sIkpIFP5qA). 

> This are just ideas. I am very happy, that I have found your project and can work with. I am looking forward to build my pellet hooper (I will do it on my own completely) and cook with it in summer.

Excellent!  You should post pictures here of your build, once you have it up and running.  I've often thought it would be fun and interesting to do a scratch build using PiFire.  

Happy grilling!

**Comment by McBlade** [comment]
> Oh that's an interesting idea, yeah, could be something we look into for future testing. If you just want to test it out on your bench, I'd recommend running PiFire with the prototype grill & adc modules installed. This will allow you to simulate the functionality, so that you can test all the features. If you're running Linux on your desktop/development machine, you can even run PiFire, using the prototype modules, on your desktop. You can run sudo bash modules.sh from the command line to configure the prototype modules. You'll have to restart the control.py script after making these changes either through supervisor or by rebooting.

I have it on an Pi in the moment and it is working. I have tested the notifications just with lower temperature notification and a hand on the probe. :D

> Notifications should work in monitor mode as well.

Hmm, ok. That was not working. It starts working in normal operation mode. I will check this on monday. Maybe I have done something wrong.

> Notifications are working for grill probes as well. Just click the bell icons in the WebUI on each of the probes. This will give you a notification once the set temperature is achieved.

Sorry for the misunderstanding. I know that the grill probe has a temperature notification, but I mean a second one. That you have one for overtemperature and one for undertemperature on the probe. ;) Could be interesting to have an alarm output on a GPIO, too. For some kind of buzzer.

> Excellent! You should post pictures here of your build, once you have it up and running. I've often thought it would be fun and interesting to do a scratch build using PiFire.

I will do. :)

**Reply by nebhead** [reply]
> Hmm, ok. That was not working. It starts working in normal operation mode. I will check this on monday. Maybe I have done something wrong.

OK, so I think there is a bug here.  Thanks for pointing this out.  I just tested with monitor mode and it seems that none of the notifications are firing.  :(  I will get that fixed and commit.  
 
> Sorry for the misunderstanding. I know that the grill probe has a temperature notification, but I mean a second one. That you have one for overtemperature and one for undertemperature on the probe. ;) Could be interesting to have an alarm output on a GPIO, too. For some kind of buzzer.

Right, all of the probes have notifications - so the food probes and the grill probes should all have little bells underneath.  

![image](https://user-images.githubusercontent.com/19876886/148658546-ccf66b25-3aff-416b-ba73-8fd70faef7e7.png)

When you click them, you can set a temperature setpoint for notification.  The picture above illustrates how this should look.

**Reply by nebhead** [reply]
> > Hmm, ok. That was not working. It starts working in normal operation mode. I will check this on monday. Maybe I have done something wrong.
> 
> OK, so I think there is a bug here. Thanks for pointing this out. I just tested with monitor mode and it seems that none of the notifications are firing. :( I will get that fixed and commit.

Fixed!  Main and Development branches are up to date.  You should be able to pull the fix by going into your pifire folder and doing a git pull.

**Reply by McBlade** [reply]
Yes, seems to be fixed. Now the notification is working.

About the temperature issue. At the moment, my pi shows following behavior. I set a temperature point for the grill probe and start heating. If the temp is reached there will be a notification and the temperature notification at the grill probe will be deleted.

I think about to have two temperatures only at the grill probe. That you get a notification if the temperature will be overshooted due to any failure or is less then the minimum. So you get an information on the most problems during the cooking.

![temperature](https://user-images.githubusercontent.com/97340424/148752466-a71d2ead-5e05-4caf-968a-ccb97de50adc.jpg)

It is just an idea. :)

**Reply by nebhead** [reply]
Yeah, I suppose we have that for the Grill Probe (safety settings) but yeah, not for the food probes.  Interesting idea. I'd say maybe a future update, we can look at something like this.  Some day... probably way in the future, I was thinking it would be good to re-architect PiFire's temperature input and notification architecture to be modular/object oriented.  In this way, it would be possible to add more grill probes / food probes / and notification options vs. the current implementation.  But it's a lot of work and would need lots of validation. :)  So, definitely a future thing.
